src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/helpers.ts

```diff
@@ -0,0 +1,113 @@
+import Decimal from 'decimal.js';
+
+export const getStakeFromSlider = (
+  sliderAmount: number,
+  remainingToStake: Decimal,
+  minUserStake: Decimal,
+) =>
+  new Decimal(sliderAmount)
+    .div(100)
+    .times(remainingToStake.sub(minUserStake))
+    .plus(minUserStake);
+
+export const getFinalStake = (
+  amount: number,
+  remainingToStake: Decimal,
+  minUserStake: Decimal,
+) => {
+  const stake = getStakeFromSlider(amount, remainingToStake, minUserStake);
+  return stake.round().toString();
+};
+
+export const convertStakeToPercentage = (
+  stake: string | Decimal,
+  requiredStake: Decimal,
+) => {
+  const divisibleRequiredStake = requiredStake.isZero() ? 1 : requiredStake;
+
+  return new Decimal(stake).div(divisibleRequiredStake).mul(100).toDP(2);
+};
+
+export const formatStakePercentage = (stakePercentage: number) => {
+  const formatted = stakePercentage.toFixed(2);
+
+  if (formatted === '1.00') {
+    return '1';
+  }
+  if (formatted === '100.00') {
+    return '100';
+  }
+
+  return formatted;
+};
+
+export const getRemainingToStake = (
+  isObjection: boolean,
+  requiredStake: Decimal,
+  { yay, nay }: { yay: string; nay: string },
+) => {
+  const remainingToFullyYayStaked = requiredStake.sub(yay);
+  const remainingToFullyNayStaked = requiredStake.sub(nay);
+
+  const remainingToStake = isObjection
+    ? remainingToFullyNayStaked
+    : remainingToFullyYayStaked;
+
+  return { remainingToStake, remainingToFullyNayStaked };
+};
+
+export const getStakedPercentages = (
+  isObjection: boolean,
+  requiredStake: Decimal,
+  {
+    yay,
+    nay,
+  }: {
+    yay: string;
+    nay: string;
+  },
```

Comment:
This is a bit confusing. It's the staked amount? Or the staked side? _(I know it's the amount, but maybe a more descriptive variable name?)_